reactor-core/src/test/java/reactor/core/publisher/FluxBufferPredicateTest.java

```diff
@@ -912,6 +916,153 @@ public void discardOnError() {
 		            .hasDiscardedExactly(1, 2, 3);
 	}
 
+	static private final Context DISCARD_RACE_WITH_ON_NEXT_CTX = Context.of(Hooks.KEY_ON_DISCARD, (Consumer<?>) (Object o) -> ((AtomicInteger) o).incrementAndGet());
+
+	BaseSubscriber<ArrayList<AtomicInteger>> createDiscardRaceWithOnNextDownstream(final AtomicLong receivedCounter) {
+		return new BaseSubscriber<ArrayList<AtomicInteger>>() {
+			@Override
+			public Context currentContext() {
+				return DISCARD_RACE_WITH_ON_NEXT_CTX;
+			}
+
+			@Override
+			protected void hookOnNext(ArrayList<AtomicInteger> value) {
+				receivedCounter.addAndGet(value.size());
+			}
+		};
+	}
+
+	@Test
+	public void discardRaceWithOnNext_bufferAdds() {
```

Comment:
this one could _really_ use a stress test instead

Reply:
no time for this unfortunately, too much on my plate. can you open a follow up issue?

Reply:
mhm... okay 😅

Reply:
(plus jcstress was added in master, this is targeting 3.2 so it would have to be done separately anyway)

Reply:
Maybe we can remove the test from this PR and add it in master later as a follow up? These really make the test execution longer and more heavyweight...

Reply:
I can do smaller loops, to at least keep a minimum of testing while not impacting the time too much?

Reply:
to be honest the worst offender seems to be pre-existing `requestRaceWithOnNextLoops` (32s)